### DISCUSSION TITLE
Support for Apple Silicon coming in 2.0?

### DISCUSSION THREAD

**Discussion by ghost** [alire-project/alire#1492] [Q&A] [answered]
Hey there,

first of all, thanks a lot for providing Alire as a very nice solution for managing Ada projects, packages and toolchains!

I've been using Alire for some time on an Intel-based Mac, but have since switched to an Apple Silicon-based one. With version 2.0 of Alire coming, I'd like to ask whether support for Apple Silicon-based Mac is planned with this new version.

Thanks in advance and best regards,

Jens

**Comment by Fabien-Chouteau** [comment]
Hello @jens-siebert, thanks for the nice feedback.

We would love to support Apple M1/2/etc. but the limiting factor so far is the lack of support for those platforms in GitHub Continuous Integration pipeline. That means we are not able to produce a release yet.

**Reply by ghost** [reply]
Hey @Fabien-Chouteau,

thanks for the quick answer! But what about these:

https://github.blog/2023-10-02-introducing-the-new-apple-silicon-powered-m1-macos-larger-runner-for-github-actions/

Best regards,

Jens

**Reply by Fabien-Chouteau** [reply]
This is indeed promising, but right now this is not available in the standard enterprise plan that we have for the Alire project.

**Reply by ghost** [reply]
I see. Anyway, thanks for the feedback!

**Reply by ghost** [reply]
@Fabien-Chouteau @mosteo FYI: https://github.blog/changelog/2024-01-30-github-actions-introducing-the-new-m1-macos-runner-available-to-open-source/

**Reply by ghost** [reply]
@Fabien-Chouteau @mosteo are there any news regarding this topic? Apple Silicon-based runners now seem to be available for open source plans as well (see my post above)...

**Reply by mosteo** [reply, marked as answer]
I'm currently focused on getting `2.0` out of the door ASAP, so unless @Fabien-Chouteau considers this a must for `2.0`, I think it will have to wait for `2.1`.

**Reply by simonjwright** [reply]
I think that provision of compilers is orthogonal to alr! Once we have [GNAT FSF Builds](https://github.com/alire-project/GNAT-FSF-builds) set up for the M1 runners (there's already a proof of concept) it should be "simply" a matter of updating the index.

In the mean time, I have built my own [support for aarch64](https://github.com/simonjwright/alire-index.mac).

**Reply by Fabien-Chouteau** [reply]
@reznikmm Is working on having compilers from GNAT FSF Builds. Once we have that it should be very easy to add Alire.

**Reply by ghost** [reply]
@mosteo @Fabien-Chouteau @simonjwright Thank you so much for your responses, I really appreciate them! I'm looking forward to this appearing in nightly builds so I can help test and probably contribute 👍

**Comment by michaelayres** [comment]
Not an answer, but I would like to converse with you a little on your set up with Alire for some time on an Intel-based Mac.
